.github/workflows/base.yml

```diff
@@ -0,0 +1,48 @@
+on:
+  push:
+    branches:
+     - "*"
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        PYSPARK_VERSION: ["2.4", "3.0", "3.1.3", "3.2"]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v2
+        name: Install Python 3.7
+        if: startsWith(matrix.PYSPARK_VERSION, '3')
+        with:
+          python-version: 3.7
+
+      - uses: actions/setup-python@v2
+        name: Install Python 3.6
+        if: startsWith(matrix.PYSPARK_VERSION, '2')
+        with:
+          python-version: 3.6
```

Comment:
I would suggest just support Python 3.8+ officially (at least in CI), Python 3.6 is EOL already and Numpy dropped support since 1.22.0 https://numpy.org/devdocs/release/1.22.0-notes.html. There's a security issue that mandate us to upgrade to 1.22+ as well.

Reply:
Thanks! I would assume those workflow should start running in this parent repo once you merge, so it's ok to keep it in the same PR. Let's drop the other legacy runners in CI.

Reply:
@chenliu0831  [Here](https://github.com/lecardozo/python-deequ/actions/runs/3068883685/jobs/4956877944#step:7:224) we can see the issue happening on spark 2.4.

Reply:
Okay.. let's keep numpy upgrade separate for now. Does PySpark 2.4 work with python 3.8+?

Reply:
Seems good to me! I'll drop it from CI and keep numpy as 1.22+